tests/integration-tests/conftest.py

```diff
@@ -1108,6 +1112,43 @@ def pytest_runtest_makereport(item, call):
     # be "setup", "call", "teardown"
     setattr(item, "rep_" + rep.when, rep)
 
+    if rep.when in ["setup", "call"] and rep.failed:
+        try:
+            update_failed_tests_config(item)
+        except Exception as e:
+            logging.error("Failed when generating config for failed tests: %s", e, exc_info=True)
+
+
+def update_failed_tests_config(item):
+    out_dir = Path(item.config.getoption("output_dir"))
+    if not str(out_dir).endswith(".out"):
+        # Navigate to the parent dir in case of parallel run so that we can access the shared parent dir
+        out_dir = out_dir.parent
+
+    out_file = out_dir / "failed_tests_config.yaml"
+    logging.info("Updating failed tests config file %s", out_file)
+    # We need to acquire a lock first to prevent concurrent edits to this file
+    with FileLock(str(out_file) + ".lock"):
+        failed_tests = {"test-suites": {}}
+        if out_file.is_file():
+            with open(str(out_file)) as f:
+                failed_tests = yaml.safe_load(f)
+
+        feature, test_id = item.nodeid.split("/", 1)
```

Comment:
it would be useful to have an example of the item.nodeid here.

Reply:
will add this in a later PR